src/diagnostics.rs

```diff
@@ -58,7 +58,7 @@ impl MachineStopType for TerminationInfo {}
 
 /// Miri specific diagnostics
 pub enum NonHaltingDiagnostic {
-    CreatedPointerTag(NonZeroU64),
+    CreatedPointerTag(NonZeroU64, Permission, AllocId, AllocRange),
```

Comment:
Now that there are so many fields, it seems better to use a struct variant and give them names and comments explaining their purpose.